llvm/lib/Transforms/Scalar/LowerMatrixIntrinsics.cpp

```diff
@@ -2194,6 +2206,49 @@ class LowerMatrixIntrinsics {
     return true;
   }
 
+  /// Lower uniform shape intrinsics, if shape information is available.
+  bool VisitUniformIntrinsic(IntrinsicInst *Inst) {
+    auto I = ShapeMap.find(Inst);
```

Comment:
If we pass ShapeInfo also to `VisitCallInsts` in https://github.com/llvm/llvm-project/pull/142487, it seems simpler to just inline the code to handle abs/fabs at the call site?